Migrate 15-check-container-id.sh to ginkgo

joestringer opened the pull request:
Fixes: #2527

Signed-off-by: Joe Stringer <joe@covalent.io>

joestringer committed:
test: Migrate 15-check-container-id.sh to ginkgo

Fixes: #2527

Signed-off-by: Joe Stringer <joe@covalent.io>

eloycoto commented on test/runtime/monitor.go:
```diff
@@ -249,4 +257,54 @@ var _ = Describe("RuntimeValidatedMonitorTest", func() {
+		cancel()
+
+		// Expected full example output:
+		// CPU 01: MARK 0x3de3947b FROM 48896 DEBUG: Attempting local delivery for container id 29381 from seclabel 263
```
Why you are not using the `ContainSubstring` function? 

You can do something like: 

```
eps := vm.GetEndpointsIds()
Expect(res).To(ContainSubstring(fmt.Sprintf("FROM %s DEBUG: Attempting local delivery for container id %s from seclabel", eps[helpers.Client], eps[helpers.Server])
```

Less logic to understand, and you're validating that the containers IDS are correct. 

Regards

joestringer replied:
I'm converting the test over pretty faithfully, so it's not only testing that the endpoints we have are in the output, but also that the output doesn't have endpoints that don't exist. That's why it's reading the endpoint IDs out of the output and performing the lookup, rather than just validating that the expected endpoints exist in the monitor output.

joestringer committed:
Ginkgo: Refactor common startup code into helpers

Shift the code to wait for the node to become ready and the network to
be created into the CreateNewRuntimeHelper() function, which is already
called in the initialize function of each test file.

Set default policy enforcement mode when getting the *SSHMeta at the
beginning of the test to make the tests more reliable and reduce
boilerplate. Tests which need other settings can explicitly modify this.

Signed-off-by: Joe Stringer <joe@covalent.io>

eloycoto reviewed (approved):
All good, and thanks for the changes in helpers.CreateNewRuntimeHelper :-)
